pkg/handlers/handlers.go

```diff
@@ -1,44 +1,99 @@
 package handlers
 
 import (
+	"errors"
 	"fmt"
 	"io"
 
 	"github.com/gabriel-vasile/mimetype"
+	"github.com/mholt/archiver/v4"
 
 	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/readers"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
 )
 
-// readSeekCloser is an interface that combines the functionality of io.ReadSeekCloser and io.ReaderAt.
-// It supports reading data, seeking within an open resource, and closing the resource once operations are complete.
-// Additionally, it allows reading from a specific offset within the resource without altering its current position,
-// enabling efficient and flexible data access patterns. This interface is particularly useful for handling files
-// or other data streams where random access and sequential processing are required.
-type readSeekCloser interface {
-	io.ReadSeekCloser
-	io.ReaderAt
+// fileReader is a custom reader that wraps an io.Reader and provides additional functionality for identifying
+// and handling different file types. It abstracts away the complexity of detecting file formats, MIME types,
+// and archive types, allowing for a more modular and extensible file handling process.
+//
+// fileReader leverages the archiver and mimetype packages for file type identification and provides information
+// about the detected file format, MIME type, and whether the file is an archive. This information can be
+// used by FileHandler implementations to make decisions on how to process the file.
+//
+// By encapsulating the file type detection logic, fileReader simplifies the implementation of FileHandler and
+// promotes a more cohesive and maintainable codebase. It also embeds a BufferedFileReader to provide efficient
+// random access to the file content.
+type fileReader struct {
+	format   archiver.Format
+	mimeType mimeType
+	*readers.BufferedFileReader
+	isArchive bool
```

Comment:
It should also be used in `archive.go` `L88-91` [here](https://github.com/trufflesecurity/trufflehog/pull/2825/files#diff-ad1adabfbf61332dbddae0d14b27a18a8f23b9f372874e1e340ebb0b2cbc54daR88-R91), sorry I should've explicitly called it out since the rest of the code was a lift and shift.